tgui/packages/tgui/interfaces/Pressurizer.js

```diff
@@ -0,0 +1,223 @@
+import { useBackend } from '../backend';
+import { Button, Divider, Flex, LabeledList, NumberInput, ProgressBar, Section } from '../components';
+import { Window } from '../layouts';
+import { PortableBasicInfo } from './common/PortableAtmos';
+
+const FanState = {
+  Off: 0,
+  In: 1,
+  Out: 2,
+};
+
+export const Pressurizer = (props, context) => {
+  const { act, data } = useBackend(context);
+
+  const {
+    airSafe,
+    blastArmed,
+    blastDelay,
+    connected,
+    emagged,
+    fanState,
+    materialsCount,
+    materialsProgress,
+    maxArmDelay,
+    maxPressure,
+    maxRelease,
+    minArmDelay,
+    minBlastPercent,
+    minRelease,
+    pressure,
+    processRate,
+    releasePressure,
+  } = data;
+
+  const handleSetPressure = releasePressure => {
+    act("set-pressure", {
+      releasePressure,
+    });
+  };
+
+  const handleSetBlastDelay = blastDelay => {
+    act("set-blast-delay", {
+      blastDelay,
+    });
+  };
+
+  const handleSetProcessRate = processRate => {
+    act("set-process_rate", {
+      processRate,
+    });
+  };
+
+  const handleSetFan = fanState => {
+    act("fan", {
+      fanState,
+    });
+  };
+
+  const sufficientPressure = () => {
+    return (pressure < (maxPressure * minBlastPercent));
+  };
```

Comment:
As this is all based on props, you could just generate once and use throughout.
```suggestion
  const hasSufficientBlastPressure = pressure < maxPressure * minBlastPercent;
```
(With associated changes to usages of `sufficientPressure()` to `hasSufficientBlastPressure`.)

Reply:
Updated as per review comment.